### PR TITLE
[Fast-reboot]: Gracefully shutdown syncd in fast-reboot

### PR DESCRIPTION
**Pull request by pavel-shirshov**
**- What I did**
Add a feature to gracefully shutdown syncd

**- How I did it**
Using syncd_request_shutdown utility

**- How to verify it**
Run fast-reboot test

**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**

-->

### CODE REVIEW

File: scripts/fast-reboot [hunk 1]
```diff
@@ -41,6 +41,18 @@ docker kill lldp > /dev/null
 # Kill teamd, otherwise it gets down all LAGs
 docker kill teamd > /dev/null
 
+# Kill swss before stopping syncd
+docker kill swss > /dev/null
+
+# Gracefully stop syncd
+docker exec -ti syncd /usr/bin/syncd_request_shutdown --cold > /dev/null
```
Comment (lguohan) [inline scripts/fast-reboot]: will this break the mellanox platform?

File: scripts/fast-reboot [hunk 2]
```diff
@@ -41,6 +41,15 @@ docker kill lldp > /dev/null
 # Kill teamd, otherwise it gets down all LAGs
 docker kill teamd > /dev/null
 
+# Gracefully stop syncd
+docker exec -ti syncd /usr/bin/syncd_request_shutdown --cold > /dev/null
+
+# Check it was stopped
+until [[ $(docker top syncd | grep -q /usr/bin/syncd) -eq 0 ]]
```
Comment (yxieca) [inline scripts/fast-reboot]: I think you need to s/until/while/

Comment (jleveque) [reply scripts/fast-reboot]: Both `grep -q` and `grep -c` will return 0 if any matches are found, and will return 1 if no matches are found.

Therefore, either are applicable, however, the condition needs to be changed to either

`until [[ $(docker top syncd | grep -q /usr/bin/syncd) -eq 1 ]]`

or 

`while [[ $(docker top syncd | grep -q /usr/bin/syncd) -eq 0 ]]`

Comment (yxieca) [reply scripts/fast-reboot]: $(...) is getting the output of the execution. Therefore, with -q it is always ''.

With -c, the output would be how many lines find. We want to wait until there is 0 lines.

Comment (jleveque) [reply scripts/fast-reboot]: You're right, Ying. For some reason I was stuck on the return value, not the stdout output.

File: scripts/fast-reboot (hunk 1 above)
Comment (yxieca) [reply scripts/fast-reboot]: I see. I missed the other change. Sorry.

Regards,
Ying

**Pull request merged**